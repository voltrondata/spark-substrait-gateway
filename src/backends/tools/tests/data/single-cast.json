{
  "relations": [
    {
      "root": {
        "input": {
          "project": {
            "input": {
              "read": {
                "common": {
                  "direct": {}
                },
                "baseSchema": {
                  "names": [
                    "r_regionkey",
                    "r_name",
                    "r_comment"
                  ],
                  "struct": {
                    "types": [
                      {
                        "i32": {
                          "nullability": "NULLABILITY_REQUIRED"
                        }
                      },
                      {
                        "string": {
                          "nullability": "NULLABILITY_REQUIRED"
                        }
                      },
                      {
                        "string": {
                          "nullability": "NULLABILITY_NULLABLE"
                        }
                      }
                    ],
                    "nullability": "NULLABILITY_REQUIRED"
                  }
                },
                "namedTable": {
                  "names": [
                    "#2"
                  ]
                }
              }
            },
            "expressions": [
              {
                "cast": {
                  "type": {
                    "fixedChar": {
                      "length": 25,
                      "nullability": "NULLABILITY_REQUIRED"
                    }
                  },
                  "input": {
                    "literal": {
                      "fixedChar": "DELIVER IN PERSON"
                    }
                  }
                }
              }
            ]
          }
        }
      }
    }
  ]
}