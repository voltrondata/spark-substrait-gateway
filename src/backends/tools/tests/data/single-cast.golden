{
  "relations": [
    {
      "root": {
        "input": {
          "project": {
            "common": {},
            "input": {
              "project": {
                "common": {
                  "emit": {
                    "outputMapping": [
                      0,
                      1,
                      2,
                      4
                    ]
                  }
                },
                "input": {
                  "read": {
                    "common": {
                      "direct": {}
                    },
                    "baseSchema": {
                      "names": [
                        "r_regionkey",
                        "r_name",
                        "r_comment"
                      ],
                      "struct": {
                        "types": [
                          {
                            "i32": {
                              "nullability": "NULLABILITY_REQUIRED"
                            }
                          },
                          {
                            "string": {
                              "nullability": "NULLABILITY_REQUIRED"
                            }
                          },
                          {
                            "string": {
                              "nullability": "NULLABILITY_NULLABLE"
                            }
                          }
                        ],
                        "nullability": "NULLABILITY_REQUIRED"
                      }
                    },
                    "namedTable": {
                      "names": [
                        "#2"
                      ]
                    }
                  }
                },
                "expressions": [
                  {
                    "literal": {
                      "fixedChar": "DELIVER IN PERSON"
                    }
                  }
                ]
              }
            },
            "expressions": [
              {
                "cast": {
                  "type": {
                    "fixedChar": {
                      "length": 25,
                      "nullability": "NULLABILITY_REQUIRED"
                    }
                  },
                  "input": {
                    "selection": {
                      "directReference": {
                        "structField": {
                          "field": 3
                        }
                      },
                      "rootReference": {}
                    }
                  }
                }
              }
            ]
          }
        }
      }
    }
  ]
}